python/hail/dataset.py

```diff
@@ -1598,21 +1598,29 @@ def export_vcf(self, output, append_to_header=None, parallel=False):
 
             We strongly recommended compressed (``.bgz`` extension) and parallel output (``parallel=True``) when exporting large VCFs.
 
-        Consider the workflow of importing VCF to VDS and immediately exporting VDS to VCF:
+        Hail exports the fields of Struct ``va.info`` as INFO fields,
+        the elements of Set[String] ``va.filters`` as FILTERS,
+        and the value of Float64 ``va.qual`` as QUAL. No other variant annotations are exported.
+        
+        The FORMAT field is generated from the genotype type, which must 
+        be :py:class:`~hail.expr.TGenotype` or :py:class:`~hail.expr.TStruct`. If the type is
+        :py:class:`~hail.expr.TGenotype`, then the FORMAT fields will be GT, AD, DP, GQ, and PL.
+        If the type is :py:class:`~hail.expr.TStruct`,
+        then the exported FORMAT fields will be the names of each field of the Struct.
+        
+        INFO and FORMAT fields may be generated from Struct fields of type Call, Int32, Float32, Float64, or String.
+        If a field has type Int64, every value must be a valid Int32.
+        Arrays and Sets containing these types are also allowed but cannot be nested;
+        for example, Array[Array[Int32]] is invalid.
+        Sets and Arrays are output with the same comma-separated format.
+        Lastly, fields of type Boolean are allowed in ``va.info`` to generate INFO fields of type Flag.
+
+        Consider the workflow of importing a VCF to VDS and immediately exporting VDS to VCF:
 
         >>> vds.export_vcf('output/example_out.vcf')
 
         The *example_out.vcf* header will contain the FORMAT, FILTER, and INFO lines present in *example.vcf*. However, it will *not* contain CONTIG lines or lines added by external tools (such as bcftools and GATK) unless they are explicitly inserted using the ``append_to_header`` option.
 
-        Hail only exports the contents of ``va.info`` to the INFO field. No other annotations besides ``va.info`` are exported.
-
-        The genotype schema must have the type :py:class:`~hail.expr.TGenotype` or :py:class:`~hail.expr.TStruct`. If the type is
-        :py:class:`~hail.expr.TGenotype`, then the FORMAT fields will be GT, AD, DP, GQ, and PL.
-        If the type is :py:class:`~hail.expr.TStruct`, then the exported FORMAT fields will be the names of each field of the Struct.
-        Each field must have a type of String, Char, Int, Double, or Call. Arrays and Sets are also allowed as long as they are not nested.
-        For example, a field with type ``Array[Int]`` can be exported but not a field with type ``Array[Array[Int]]``.
-        Nested Structs are also not allowed.
```

Comment:
what happens to nested structs now?

Reply:
They are not allowed, but in my updated docs I tried to specify exactly what is allowed rather than list everything that is not allowed.

Reply:
got it---that makes sense.